components/style/properties/properties.mako.rs

```diff
@@ -1091,6 +1098,10 @@ impl PropertyDeclaration {
                     match input.try(|i| CSSWideKeyword::parse(context, i)) {
                         Ok(CSSWideKeyword::InheritKeyword) => {
                             % for sub_property in shorthand.sub_properties:
+                                if properties_seen.get_${property.ident}() {
```

Comment:
> The idea was to keep the properties_seen code next to the result_list.push occurences, since they kinda sorta operate on the same data.

Makes sense.

> As for the non-wide keywords, I missed that one - that's the only branch where pushing to the result_list is not explicit. I'll fix that.

Thanks!

> Also, I believe I've made a mistake - I'm pretty sure the line you commented on (and some lines below) should have a sub_property.ident instead of property.ident. Is this correct?

Indeed, you want `sub_property.ident` there. I don't think that should compile, what code does that generate? (you can look at it looking at `target/debug/build/style-<hash>/out/properties.rs`)